cmd/lnd/main.go

```diff
@@ -34,9 +34,20 @@ func main() {
 	// Call the "real" main in a nested manner so the defers will properly
 	// be executed in the case of a graceful shutdown.
 	if err = lnd.Main(
-		loadedConfig, lnd.ListenerCfg{}, shutdownInterceptor,
+		loadedConfig, lnd.ListenerCfg{}, *shutdownInterceptor,
 	); err != nil {
 		_, _ = fmt.Fprintln(os.Stderr, err)
 		os.Exit(1)
 	}
+	defer func() {
+		fmt.Println("Shutdown complete")
+		err := loadedConfig.LogWriter.Close()
+		if err != nil {
+			_, _ = fmt.Fprintln(os.Stderr, err)
+		}
+
+		if shutdownInterceptor.GetExitCode() == 1 {
+			os.Exit(1)
```

Comment:
Seems we can just pass the result of the exit code directly here?

Reply:
Ah yes, I was actually kept it as a `bool` first later changed to `int`. We can surely pass it directly